src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Simplification/Simplifiers/CastSimplifier.cs

```diff
@@ -81,6 +81,14 @@ public static bool IsUnnecessaryCast(CastExpressionSyntax cast, SemanticModel se
             if (IsRemovableBitwiseEnumNegation(cast, semanticModel, cancellationToken))
                 return true;
 
+            // Special case for converting a method group to object. The compiler issues a warning if the cast is removed:
+            // warning CS8974: Converting method group 'ToString' to non-delegate type 'object'. Did you intend to invoke the method?
+            if (semanticModel.GetOperation(cast.Expression, cancellationToken)?.Kind == OperationKind.MethodReference &&
+                semanticModel.GetTypeInfo(cast.Type, cancellationToken).Type is { SpecialType: SpecialType.System_Object })
```

Comment:
```suggestion
                semanticModel.GetTypeInfo(cast.Type, cancellationToken).Type?.SpecialType is SpecialType.System_Object)
```

(i'm not a fan of an entire property pattern aroudn a single property for the purpose of a null check.  i feel like `?.` is the idiomatic way to alerady do that.